### ISSUE TITLE
await createPool leads to odd behaviour

### ISSUE THREAD

**Issue by smdufb** [sidorares/node-mysql2#909]
await'ing non promises is technically valid code. But the following code behaves strangely.

`async query(qry) {
        if (!this.pool) {
            this.pool = mysql.createPool({
                connectionLimit: 10,
                host: "*****",
                user: "*****",
                password: "*****",
                database: "*****"
            });
        }
        return this.pool.query(qry).catch(console.log);
    }`

The whole thing works. But instead of queuing requests it creates a new connection for every query.

**Comment by sidorares** [comment]
looks like you are creating new pool for each query instead of reusing single pool

**Comment by sidorares** [comment]
oh, sorry, I see you assign it to a variable

**Comment by smdufb** [comment]
Yeah excuse the formatting issue. Fixed that.

**Comment by smdufb** [comment]
Also just inserted the actual 'await' now.

**Comment by sidorares** [comment]
Is `mysql` coming from `require('mysql2/promise')? "Promisified" pool returns promise-wrapped connection object

**Comment by smdufb** [comment]
Yes I'm using `require('mysql2/promise')`. But `createPool` doesn't return a promise in either mysql2 or mysql2/promise, right?

**Comment by sidorares** [comment]
correct, but awaiting non-promise should in theory give you original value

**Comment by smdufb** [comment]
Yes exactly. And it does to some extent in that the pool is created and works. Only it creates connections for every query even past the `connectionLimit`
